components/global/mixer-adapter/pkg/wso2spadapter/wso2spadapter.go

```diff
@@ -60,53 +52,94 @@ type (
+			adapter.logger.Error("error unmarshalling adapter config: ", err.Error())
 			return nil, err
 		}
 	}
 
-	buffer.WriteString(fmt.Sprintf("HandleMetric invoked with:\n  Adapter config: %s\n  Instances: %s\n",
-		cfg.String(), instances(r.Instances)))
-
-	if cfg.FilePath == "" {
-		wso2SpAdapter.logger.Debug(buffer.String())
-	} else {
-		_, err := os.OpenFile("out.txt", os.O_RDONLY|os.O_CREATE, 0666)
-		if err != nil {
-			wso2SpAdapter.logger.Error("error creating file: ", err.Error())
-			return nil, err
-		}
-		file, err := os.OpenFile(cfg.FilePath, os.O_APPEND|os.O_WRONLY, 0600)
-		if err != nil {
-			wso2SpAdapter.logger.Error("error opening file for append: ", err.Error())
-			return nil, err
-		}
+	//buffer.WriteString(fmt.Sprintf("HandleMetric invoked with:\n  Adapter config: %s\n  Instances: %s\n",
+	//	cfg.String(), getInstances(r.Instances)))
+	//
+	//if cfg.FilePath != " " {
+	//	_, err := os.OpenFile("out.txt", os.O_RDONLY|os.O_CREATE, 0666)
+	//	if err != nil {
+	//		adapter.logger.Error("error creating file: ", err.Error())
+	//		return nil, err
+	//	}
+	//	file, err := os.OpenFile(cfg.FilePath, os.O_APPEND|os.O_WRONLY, 0600)
+	//	if err != nil {
+	//		adapter.logger.Error("error opening file for append: ", err.Error())
+	//		return nil, err
+	//	}
+	//
+	//	defer func() {
+	//		err := file.Close()
+	//		if err != nil {
+	//			adapter.logger.Warn("Could not close the file")
+	//		}
+	//	}()
+	//
+	//	if _, err = file.Write(buffer.Bytes()); err != nil {
+	//		adapter.logger.Error("error writing to file: ", err.Error())
+	//		return nil, err
+	//	}
+	//}
+
+	var instances = r.Instances
+	for _, inst := range instances {
+		var attributesMap = decodeDimensions(inst.Dimensions)
+		adapter.logger.Debugf("received request : %s", attributesMap)
+		adapter.metricsPublisher.publishMetrics(attributesMap, adapter.logger, adapter.httpClient, adapter.spServerUrl) // ToDO : get the boolean value from the function to check whether the metric is delivered or not
+	}
 
-		defer file.Close()
+	return &v1beta1.ReportResult{}, nil
+}
 
-		if _, err = file.Write(buffer.Bytes()); err != nil {
-			wso2SpAdapter.logger.Error("error writing to file: ", err.Error())
-			return nil, err
-		}
+/* Send metrics to sp server */
+func (publisher SPMetricsPublisher) publishMetrics(attributeMap map[string]interface{}, logger *zap.SugaredLogger, httpClient *http.Client, spServerUrl string) bool {
+	jsonValue, _ := json.Marshal(attributeMap)
+	req, err := http.NewRequest("POST", spServerUrl, bytes.NewBuffer(jsonValue))
+	if req != nil {
+		req.Header.Set("Content-Type", "application/json")
 	}
 
-	return &v1beta1.ReportResult{}, nil
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer func() {
+		err := resp.Body.Close()
+		if err != nil {
+			logger.Warn("Could not close the body")
+		}
+	}()
+	body, _ := ioutil.ReadAll(resp.Body)
+	logger.Info("response : ", string(body))
```

Comment:
This should be a debug log and it should be something more meaningful.

Suggestion :- "Received response from SP Worker: " + string(body)